### DISCUSSION TITLE
Offline packages.

### DISCUSSION THREAD

**Discussion by bobjoe12131** [gleam-lang/gleam#3711] [Questions & support] [answered]
I am sometimes at a place where i can not access the internet. 
When i create a new project, gleam wants to access hex and get the stdlib and gleeunit package, but when it fails it quits, even though there are packages in the cache.

I found (1) <https://hexdocs.pm/hex/Mix.Tasks.Hex.Registry.html>
and someone helping me found (2) https://hex.pm/docs/mirrors

I made a local registry as in (1) at port 443 and set HEX_MIRROR to localhost as in (2) but it did not work. I also tried using /etc/hosts to set repo.hex.pm to 127.0.0.1.

I will transfer the logs later for the /etc/hosts attempt.

The main issue is using gleam while offline. There are multiple possible solutions like using the cache as a fallback, or supporting changing the repo access url.

**Comment by bobjoe12131** [comment]
DEBUG download_deps: locking_build_directory path="/home/bobjoe/building/gleam/off_test/build/packages"
TRACE download_deps: reading_file path="/home/bobjoe/building/gleam/off_test/gleam.toml"
DEBUG download_deps: manifest_not_present
  Resolving versions
 INFO download_deps: resolving_versions
DEBUG download_deps: looking_up_hex_package package="gleeunit"
TRACE download_deps: checkout waiting for idle connection: ("https", repo.hex.pm)
DEBUG download_deps: starting new connection: https://repo.hex.pm/    
TRACE download_deps: Http::connect; scheme=Some("https"), host=Some("repo.hex.pm"), port=None
DEBUG resolving host="repo.hex.pm"
DEBUG download_deps: connecting to 127.0.0.1:443
DEBUG download_deps: connected to 127.0.0.1:443
TRACE download_deps: checkout dropped for ("https", repo.hex.pm)
ERROR Failed error=DependencyResolutionFailed("Unable to determine package versions: error sending request for url (https://repo.hex.pm/packages/gleeunit)")
error: Dependency resolution failed

An error occurred while determining what dependency packages and
versions should be downloaded.
The error from the version resolver library was:

Unable to determine package versions: error sending request for url
(https://repo.hex.pm/packages/gleeunit)

**Comment by lpil** [comment, marked as answer]
There's an issue for this here: https://github.com/gleam-lang/gleam/issues/2162

**Reply by bobjoe12131** [reply]
Oh, didnt find that issue. Thanks.
